ssl/record/dtls1_bitmap.c

```diff
@@ -13,76 +13,49 @@
 /* mod 128 saturating subtract of two 64-bit values in big-endian order */
 static int satsub64be(const unsigned char *v1, const unsigned char *v2)
 {
-    int ret, i;
-
-    if (sizeof(long) == 8)
-        do {
-            const union {
-                long one;
-                char little;
-            } is_endian = {
-                1
-            };
-            long l;
-
-            if (is_endian.little)
-                break;
-            /* not reached on little-endians */
-            /*
-             * following test is redundant, because input is always aligned,
-             * but I take no chances...
-             */
-            if (((size_t)v1 | (size_t)v2) & 0x7)
-                break;
-
-            l = *((long *)v1);
-            l -= *((long *)v2);
-            if (l > 128)
-                return 128;
-            else if (l < -128)
-                return -128;
-            else
-                return (int)l;
-        } while (0);
-
-    ret = 0;
-    for (i=0; i<7; i++) {
-        if (v1[i] > v2[i]) {
-            /* v1 is larger... but by how much? */
-            if (v1[i] != v2[i] + 1)
+    int ret;
+
+    if (sizeof(long) == 8) {
+        unsigned long l1, l2;
+
+        n2l8(v1, l1);
+        n2l8(v2, l2);
+
+        ret = l1 - l2;
```

Comment:
Ick, yes. `ret` was a long in my head when I wrote that line.... but obviously not in reality. What did I say about test cases only testing the corner cases I *knew* about and had thus already coped with?

Given that `uint64_t` is available, I'll fix the code to simply use that. There's no need for a fix in 1.0.2 further than the one already present in HEAD, and submitted for PR #1387 ... although I'm still a little unhappy with the inconsistency w.r.t. wrap-around, even though I think it's entirely theoretical since DTLS sequence numbers are 48-bit.